ui-modules/utils/catalog-deleter/catalog-deleter.js

```diff
@@ -0,0 +1,145 @@
+            mode: '@',
+            symbolicName: '<',
+            version: '<',
+            onDeleting: '&',
+            onDeleted: '&',
+            onFailed: '&',
+            onDeletingFinished: '&',
+        },
+        template: template,
+        controller: ['$scope', catalogDeleterController],
+        controllerAs: 'vm',
+    };
+
+    function catalogDeleterController($scope) {
+        let vm = this;
+
+        $scope.id = $scope.symbolicName + ':' + $scope.version;
+
+        function getBundle(bundleSymbolicName, bundleVersion) {
+            catalogApi.getBundle(bundleSymbolicName, bundleVersion).then(data => {
+                $scope.bundle = data;
+
+            }).catch(err => {
+                console.log("Error loading bundle: ", err);
+                $scope.bundleError = true;
+            }).finally(() => {
+                $scope.bundleLoading = false;
+            });
+        }
+
+        if ($scope.mode==='bundle') {
+            $scope.bundleLoading = true;
+            getBundle($scope.symbolicName, $scope.version);
+
+        } else {
+            $scope.bundleLoading = true;
+            catalogApi.getType($scope.symbolicName, $scope.version).then(data => {
+                let bundleSymbolicName, bundleVersion;
+                if (data.containingBundle) {
+                    let parts = data.containingBundle.split(':');
+                    if (parts.length>=1) {
+                        bundleSymbolicName = parts[0];
+                        if (parts.length>=2) {
+                            bundleVersion = parts[1];
+                            if (parts.length>2) {
+                                throw 'Invalid containing bundle '+data.containingBundle;
+                            }
+                        }
+                    }
+                }
+                if (!bundleSymbolicName) {
+                    throw 'Unavailable or invalid containing bundle '+data.containingBundle;
+                }
+
+                getBundle(bundleSymbolicName, bundleVersion);
+
+            }).catch(err => {
+                console.log("Error loading type: ", err);
+                if ($scope.mode==='location') {
+                    // don't display an error, probably it is a legacy-installed location
+                } else {
+                    $scope.bundleError = true;
+                }
+                $scope.bundleLoading = false;
+            });
+        }
+
+        vm.delete = () => {
+            if ($scope.onDeleting) $scope.onDeleting();
+            let promise;
+            if ($scope.mode==='bundle') {
+                promise = catalogApi.deleteBundle($scope.symbolicName, $scope.version)
+            } else if ($scope.mode==='location') {
+                promise = catalogApi.deleteLocation($scope.symbolicName, $scope.version)
+            } else if ($scope.mode==='type') {
+                // not used
+                throw 'deleteType not supported';
+            } else {
```

Comment:
How about restricting deletion with `ng-if`, so that button is available/visible for locations and bundles?

Reply:
the caller controls the button, this simply control the popup, which i think is right for maximum reusability